[5.4] Improve Pre-Update Check for Joomla ‎6.0.0: fix confusing message + additional notices

cyrez opened the pull request:
Pull Request for Issue reported on Mattermost and forum: confusing message about the compatibility plugin for Joomla 5 before updating to J6.
In addition, it adds a notice message for each requirement that is not met.

> [!NOTE]
> PR updated with a new proposal.
> Update 2: new language strings (Action required instead of Error + integration of @brianteeman proposal https://github.com/joomla/joomla-cms/pull/46326 for unchecked state )

### Summary of Changes
- Displays only translated plugin name
- Use `Enabled` — `Disabled` instead of `Yes` — `No`
- Set a $option->text for badge text in checked column

### Testing Instructions
- Test on a Joomla 5.4 install
- Check that the Behaviour -  Backward Compatibility is enabled
- In Joomla updater options, set 'Update Channel' on 'Next'
- Check for Joomla update
- Control Pre-Update Check for Joomla ‎6.0.0 'Require Settings'.

- Apply patch
- Check whether the pre-update information about the compatibility plugin is clearer.

### Actual result BEFORE applying this Pull Request
Confusing text:

> The 'Behaviour - Backward Compatibility' plugin is disabled `No`
> Go to 'System - Manage - Plugins' and disable the plugin

<img width="1822" height="530" alt="Screenshot 2025-10-19 at 18-41-00 Joomla Update - W3biC - Administration" src="https://github.com/user-attachments/assets/3d7df588-8341-4fd6-ad80-eb7da7c79489" />



### Expected result AFTER applying this Pull Request
Clear statement text:

> Behaviour - Backward Compatibility `Action Required` (if not ok)
> Go to 'System - Manage - Plugins' and disable the plugin

**Not OK state:**

<img width="1818" height="618" alt="Screenshot 2025-10-19 at 18-56-46 Joomla Update - DEV iCagenda J5 - Administration" src="https://github.com/user-attachments/assets/afa3164e-8519-4e08-abbb-4043c2721942" />


 
**OK state:**

<img width="1822" height="488" alt="Screenshot 2025-10-19 at 18-32-24 Joomla Update - DEV iCagenda J5 - Administration" src="https://github.com/user-attachments/assets/9fcf461e-6f21-45c7-8ee0-2a2ae31a6f9e" />


#### Full list with additional notices

All is NOT OK:


<img width="1790" height="1770" alt="Screenshot 2025-10-21 at 17-15-17 Joomla Update - DEV iCagenda J5 - Administration" src="https://github.com/user-attachments/assets/c8ec720b-2979-4296-9c98-ffbc0fc7289b" />



All is OK:

<img width="1794" height="1230" alt="Screenshot 2025-10-19 at 18-33-09 Joomla Update - DEV iCagenda J5 - Administration" src="https://github.com/user-attachments/assets/ce632bc4-408f-469a-8aae-f9a693920f9c" />



### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: add recover information from https://manual.joomla.org/migrations/44-50/compat-plugin/#why-should-you-try-to-disable-the-plugin to https://manual.joomla.org/migrations/54-60/compat-plugin/<link>
- [ ] No documentation changes for manual.joomla.org needed

cyrez committed:
Improve UX: confusing message about the compatibility plugin

cyrez committed:
Improve UX: confusing message about the compatibility plugin

richard67 commented:
So while without your PR the information if the check result is good or bad was visible by colours green and red and the text, yes is alway good and no always bad.

Now with this PR only the color shows if the result is good or bad.

In my opinion that's an accessibility flaw.

brianteeman commented:
See #46326 for an alternative proposal

cyrez committed:
FIX UX/UI: Confusing information regarding the required settings for pre-checking the update to the next major version.

cyrez committed:
FIX UX/UI: Confusing information regarding the required settings for pre-checking the update to the next major version.

cyrez committed:
Update preupdatecheck.php

brianteeman commented:
I do not agree with this. It is not an error. The column header is checked so saying it's value is error is wrong

richard67 commented:
So we have 3 alternative PRs for the same thing, this one here, #46326 by @brianteeman and #46327 by @chmst .

cyrez commented:
Updated PR

Note for @brianteeman : it's possible to integrate your new language strings when the plugin compat for J5 is enabled only (and keep it simple when it's OK)?

cyrez commented:
> I do not agree with this. It is not an error. The column header is checked so saying it's value is error is wrong

I've tried with existing language strings, but what could be a better wording? "Warning", "Failed", "incorrect"...
As this label should work for all Requirements states.

cyrez committed:
Add action required message to Joomla update language file

cyrez committed:
Update plugin label based on compatibility state

cyrez committed:
Update error message for pre-update check

cyrez commented:
@brianteeman i've update with a new language string "Action required" instead of error, and integration of your 2 new language strings.

In the description of this PR, i've added 2 screenshots: When checked is ok and when it needs action.

Requirement checked:

<img width="911" height="273" alt="Capture d’écran 2025-10-19 à 14 20 02" src="https://github.com/user-attachments/assets/22e58199-aa33-41cb-8cb7-03922d447be8" />

Requirement not checked:

<img width="899" height="289" alt="Capture d’écran 2025-10-19 à 14 19 43" src="https://github.com/user-attachments/assets/0de1bc76-4aaf-444b-a36a-19925ae66815" />

cyrez committed:
Add new error messages for database and PHP issues

cyrez committed:
Refactor PHP version and module checks in UpdateModel

cyrez commented:
Larger update of this PR with new language strings and notices for each requirements.

When all is OK:
<img width="1788" height="1226" alt="Screenshot 2025-10-19 at 16-38-16 Joomla Update - DEV iCagenda J5 - Administration" src="https://github.com/user-attachments/assets/005623b0-f5e9-4a05-a603-d4e244eb0738" />

When all is NOT OK (for preview of each one):
<img width="1752" height="1702" alt="Screenshot 2025-10-19 at 16-36-25 Joomla Update - DEV iCagenda J5 - Administration" src="https://github.com/user-attachments/assets/3386dfab-b38e-4a42-a9ae-3f899b134e2b" />

cyrez committed:
Update plugin BC disabled notice link in language file

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,6 +3,7 @@
 ; License GNU General Public License version 2 or later; see LICENSE.txt
 ; Note : All ini files need to be saved as UTF-8
 
+COM_JOOMLAUPDATE_ACTION_REQUIRED="Action required"
```
```suggestion
COM_JOOMLAUPDATE_ACTION_REQUIRED="Action Required"
```

cyrez replied:
Thank you @brianteeman !
I realize that I mixed upper and lower case letters in the new strings, and I don't know the rule about that... But consistency should definitely be reviewed.

richard67 replied:
Rule is that titles or labels are capitalised, see https://manual.joomla.org/docs/next/user-interface-text/capitalisation/ .

cyrez replied:
Thank you @richard67 
So i've udpated the language file with Capitalisation for labels as mentionned in the manual.

brianteeman commented:
Links to manual are only ever in english but if you link to docs then they are in the current language if they are in the format https://docs.joomla.org/Special:MyLanguage/LINK

cyrez committed:
Update administrator/language/en-GB/com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

cyrez commented:
> Links to manual are only ever in english but if you link to docs then they are in the current language if they are in the format https://docs.joomla.org/Special:MyLanguage/LINK

Do you mean for "Required Settings" or for all links? As not sure all exist on docs.joomla.org

cyrez commented:
I think i will use badge as @bembelimen suggested, for consistency with other parts of the pre-update process.

cyrez committed:
Update icons and classes for preupdatecheck.php

cyrez committed:
Update plugin BC disabled notice for compatibility

cyrez committed:
[Label Capitalisation] Update error messages and notices in language file

cyrez commented:
> FYI some of the checks will never be displayed as failed. For example the php version check will never display a fail as the component checks that before this page even loads

Yes, i saw that as well. But for consistency, i think it's best to keep the new language strings for it, or no?

cyrez commented:
About links, as @brianteeman i think it would be best to link to the User documentation instead of the manual, and with possible translation.
As far as i know, a new User Documentation site is about to be launched, so this could be an update to plan for those strings when the new user documentation site will be public, and with the needed information there?

cyrez committed:
Enhance language strings in com_joomlaupdate.ini

Updated various text strings in the Joomla update component language file to include additional information and improve clarity.

cyrez commented:
After talk with @chmst, I've improved accessibility for links with a target blank. All those external links in language strings (not only the ones related to this PR, but as i've got the hands on it, let's fix it in the same time) missed a title attribute for screen readers.

That does not change the main purpose of this PR, but improve all com_joomlaupdate language accessibility in one shot.

Thank you @chmst for your help! 👍

cyrez commented:
> being wrong on other places is not a reason to repeat the error. what you had before was perfectly ok

What i did at first is my favorite one as well.

I will ask the A11y team to have a view on it. ;-)

cyrez reviewed (commented):
Previous solution was better for accessibility with a screen reader.

cyrez commented:
> @cyrez Is this PR ready for testing now or do you expect more changes?

Not yet, i'm just reverting the language strings with update for accessibility with a screen reader (did all the test, and will set a working solution).
So, just wait a few minutes, and it will be ok for testing ;-)

cyrez committed:
Update English language strings in com_joomlaupdate

cyrez committed:
Add only new strings to be patch version ready

Other improvements to be done in a separated PR

cyrez committed:
Missing one reverted string to visually hidden text instead of title attribute

brianteeman commented:
Thanks for correcting the accessibility of the links

I am still not convinced that https://manual.joomla.org/docs/next/ is the correct site to link to at the moment due to it being english only

ChristineWk commented:
I have tested this item :white_check_mark: successfully on 9e5637526c24320d32afedf5ffc1876293954200<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46324">issues.joomla.org/tracker/joomla-cms/46324</a>.</sub>

dautrich commented:
I have tested this item :white_check_mark: successfully on 9e5637526c24320d32afedf5ffc1876293954200<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46324">issues.joomla.org/tracker/joomla-cms/46324</a>.</sub>

cyrez committed:
Revise plugin notices and deprecations in language file

Updated plugin notices and deprecated messages for Joomla update.

richard67 commented:
What seems inconsistent to me (at least in the screenshots in the description):
- When the condition for the Joomla 5 b/c plugin fails, the title is "The 'Behaviour - Backward Compatibility' Plugin Must be Disabled".
- When the condition for the Joomla 6 b/c plugin fails, the title is "Behaviour - Backward Compatibility 6"

For consistency I would expect the latter to be: "The 'Behaviour - Backward Compatibility 6' Plugin Must be Enabled".

cyrez commented:
Note: i can add those strings for now.
But maybe the global logic can be reviewed in the case no third-party extensions could break the site on J6. (by security, to not block update only if zero third-party extensions installed)?

cyrez committed:
Update plugin compat 6 label based on state

cyrez committed:
Add label for enabling the compatibility 6 plugin when disabled

dautrich commented:
I have tested this item :white_check_mark: successfully on d0508b62e20edd3c2375f66aca832fdd404e9de0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46324">issues.joomla.org/tracker/joomla-cms/46324</a>.</sub>

richard67 commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46324">issues.joomla.org/tracker/joomla-cms/46324</a>.</sub>

muhme committed:
Merge branch '5.4-dev' into patch-61

muhme commented:
✅ Final test before merge with JBT 5.4-dev
  * Before the PR, seen the confusing double Negation
    `'Behaviour - Backward Compatibility' plugin is disabled No`
    with enabled 'Behaviour - Backward Compatibility' plugin 
  * Applied PR with `gh pr checkout 46324`
    * Checked everything red and the suggestions and links are correct
      * Set `mbstring.language = Japanese` in `php.ini` file and checked the recommended `.htaccess` fix is working
      * Enabled 'Behaviour - Backward Compatibility' and disabled 'Behaviour - Backward Compatibility 6'
      * Remaining tests set to false by *hacking* `-> state = false;`
      * Links are going to Requirements for Joomla! 6.x
      * ⚠️ Not to disable `parse_ini_string` PHP function is not named in Requirements for Joomla! 6.x -> I will handle this in parallel
      * ⚠️ [54-60/compat-plugins](https://manual.joomla.org/migrations/54-60/compat-plugin/) is not up-to-date -> I will handle this together with 6.0 RMs
    * Checked everything is green
    * Tested in light and dark mode
